lib/SPIRV/libSPIRV/spirv.hpp

```diff
@@ -71,21 +71,18 @@ enum ExecutionModel {
     ExecutionModelFragment = 4,
     ExecutionModelGLCompute = 5,
     ExecutionModelKernel = 6,
-    ExecutionModelCount, /* internal use only */
```

Comment:
I took this approach before, i.e. defining my own enum types matching the enum values in spirv.hpp. Later I switched to use enum types in spirv.hpp since there is lots of maintenance burden due to changes in spirv.hpp. Things may be easier now since SPIR-V spec is finalized. 

Using a script to generate our own enum definition based on JSON may be a good idea, though it will take some efforts.

Another approach, instead of generate home-brew enum types, we use script to generate isValid() function based on JSON, and the function will be like:

    isValid(Enum E) {
      switch(E) {
        case Enum1:
        case Enum2:
        //... all defined enum values
           return true;
        default:
           return false
     }